src/lib/ndpi_utils.c

```diff
@@ -2295,6 +2295,26 @@ float ndpi_entropy(u_int8_t const * const buf, size_t len) {
   return entropy;
 }
 
+/* ******************************************************************** */
+
+u_int16_t ndpi_calculate_icmp_checksum(u_int8_t const * const buf, size_t len) {
```

Comment:
I didn't checked, but this algorithm seems identical to the TCP/UDP ones. **If** that is true, should we use a more generic name for the function (especially since we export it to the public API)?